application/Common/Shared/Utility/NativeMethods.cs

```diff
@@ -22,18 +32,23 @@ public static class NativeMethods
         /// </summary>
         public static void DoWin32MessageLoop()
         {
-            int status;
-            while ((status = GetMessage(out var msg, IntPtr.Zero, 0, 0)) != 0)
+            // We set allowsQueue to true as we expect a call to this method to be purposeful.
+            shouldCancel = true;
+            while (shouldCancel)
```

Comment:
Could you invert the boolean? It is a bit unintuitive to set `shouldCancel` to `false` to actually cancel the iteration